builder/views.py

```diff
@@ -110,28 +110,35 @@ def _get(self, request: Request, course: Course) -> Dict[str, Any]:
         return obj
 
     @login_required_method(redirect_url=None)
-    def get(self, request: Request, key: str, **kwargs) -> HttpResponse:
-        course = get_object_or_404(Course, key=key)
+    def get(self, request: Request, key: Optional[str] = None, remote_id: Optional[int] = None, **kwargs) -> HttpResponse:
+        if key:
+            course = get_object_or_404(Course, key=key)
+        else:
+            course = get_object_or_404(Course, remote_id=remote_id)
 
         if not course.has_access(request, Permission.READ):
             return JsonResponse({"success": False, "error": f"No access to instance {course.remote_id}"})
 
         return JsonResponse(self._get(request, course))
 
     @login_required_method(redirect_url=None)
-    def post(self, request: Request, key: str, **kwargs) -> HttpResponse:
+    def post(self, request: Request, key: Optional[str] = None, remote_id: Optional[int] = None, **kwargs) -> HttpResponse:
         if not request.POST:
             return JsonResponse({"success": False, "error": "No POST parameters given"})
 
-        if Course.objects.filter(key=key).exists():
-            return HttpResponse("Course already exists", status=400)
+        if key and Course.objects.filter(key=key).exists():
+            return HttpResponse(f"Course with key '{key}' already exists", status=400)
+        elif remote_id and Course.objects.filter(remote_id=remote_id).exists():
+            return HttpResponse(f"Course with id '{remote_id}' already exists", status=400)
 
         response = self._check_access(request)
         if response is not None:
             return response
 
-        if request.POST.get("key") != key:
+        if key and request.POST.get("key") != key:
             return HttpResponse("Key in POST params does not match key in URL", status=400)
+        elif remote_id and int(request.POST.get("remote_id")) != remote_id:
```

Comment:
Does this crash when the POST data does not contain any `remote_id`?

`request.POST.get("remote_id")` returns `None` when there is no parameter.
`int(None)` crashes.
`TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'`

Reply:
Yes, this does crash. There is a related issue in the put method. I can push a fix

Reply:
Should be fixed by 66a781c424121500e4d5c93cb68d6729d756d1f8